site/src/utils/formatDate.ts

```diff
@@ -0,0 +1,24 @@
+export const formatDate = (date: Date): string => {
+  if (isNaN(date.getTime())) {
+    return "To be seen";
+  } else {
+    const options: Intl.DateTimeFormatOptions = {
+      day: "numeric",
+      month: "long",
+      year: "numeric",
+    };
+
+    const formatter = new Intl.DateTimeFormat(undefined, options);
+    const formattedDateParts = formatter.formatToParts(date);
+
+    const monthNumber = date.getMonth() + 1;
```

Comment:
Suspect we can simplify this code, the formatter should be able to do this for us..
for day & month and then quarter can be calculated ourselves (like you have done)